### PR TITLE
Only run stack if a stack.yaml config is found

### PR DESCRIPTION
**Pull request by destiny-index**
It is necessary to check for a stack.yaml file to distinguish between
cabal-only projects or stack projects (which are also cabal projects
since stack is built on top of cabal).

### CODE REVIEW

**Comment by w0rp** [comment]
I was saying in a previous pull request that this would probably come up.

File: autoload/ale/handlers/haskell.vim [hunk 1]
```diff
@@ -1,5 +1,15 @@
...
 " Description: Error handling for the format GHC outputs.
+"
+function! ale#handlers#haskell#GetStackExecutable(bufnr) abort
+    if ale#path#FindNearestFile(a:bufnr, 'stack.yaml') isnot# ''
+        return 'stack'
+    else
+        " if there is no stack.yaml file, we don't use stack even if it exists,
+        " so we return '', because executable('') apparently always fails
+        return ''
+    endif
```
Comment (w0rp) [inline autoload/ale/handlers/haskell.vim]: Move `return ''` to the end of the function, instead of putting it inside `else`.

Comment (destiny-index) [reply autoload/ale/handlers/haskell.vim]: Fixed

**Comment by RyanSquared** [comment]
I'm not confident in the tests. I think that the tests should create `stack.yml`, and then in the negative tests, remove it.

**Pull request merged**